Library/Homebrew/dependency.rb

```diff
@@ -55,14 +56,29 @@ def installed?(minimum_version: nil)
     return false unless formula
 
     if minimum_version.present?
-      formula.any_version_installed? && (formula.any_installed_version.version >= minimum_version)
+      installed_version = formula.any_installed_version
+      return false unless installed_version
+
+      # Tabs prior to 4.1.18 did not have revision or pkg_version fields.
+      # As a result, we have to be more conversative when we do not have
+      # a minimum revision from the tab and assume that if the formula has a
+      # the same version and a non-zero revision that it needs upgraded.
+      if minimum_revision.present?
+        minimum_pkg_version = PkgVersion.new(minimum_version, minimum_revision)
+        installed_version >= minimum_pkg_version
+      elsif installed_version.version == minimum_version
+        formula.revision.zero?
```

Comment:
I think this may be causing current CI dependent failure when only revision bumping like in https://github.com/Homebrew/homebrew-core/actions/runs/6856551968/job/18644302924?pr=154195#step:3:244

Dependent test tries to install bottle for formula that was built in PR. Not sure if just checking `formula.latest_version_installed?` works or if there is better option.

Reply:
Can you narrow this down to the specific line and, ideally, open a PR (even if it's not fully working yet)?

The `formula.revision.zero?` part at least is essential here: that's basically the point of all my changes to this PR.

Reply:
Fixed hopefully in https://github.com/Homebrew/brew/pull/16220

Reply:
It should be that line with reproduction by local source-building a formula with extra revision bump then trying to install deps, e.g.
```console
$ brew bump-revision --write-only python-click
$ brew install -s python-click
$ brew install --only-dependencies airshare
...
==> Downloading https://ghcr.io/v2/homebrew/core/python-click/manifests/8.1.7_1-1
curl: (22) The requested URL returned error: 404

==> Pouring python-click--8.1.7.sonoma.bottle.1.tar.gz
Error: /usr/local/Cellar/python-click/8.1.7_1 is not a directory
```

Verifying latest version/revision is installed does make error go away, e.g.
```suggestion
      elsif installed_version.version == minimum_version
        formula.revision.zero? || formula.latest_version_installed?
```